File: app/src/main/java/kiwi/slang/ColorsBackground.java
package kiwi.slang;

import android.graphics.Color;

import java.util.Random;

public class ColorsBackground {

    private String[] colors = {
            "#39add1", // light blue
            "#3079ab", // dark blue
            "#c25975", // mauve
            "#e15258", // red
            "#f9845b", // orange
            "#838cc7", // lavender
            "#7d669e", // purple
            "#53bbb4", // aqua
            "#51b46d", // green
            "#e0ab18", // mustard
            "#637a91", // dark gray
            "#f092b0", // pink
            "#b7c0c7"  // light gray
    };

    int getColor() {
        Random slangGenerator = new Random();
        int randomNumber = slangGenerator.nextInt(colors.length);
        int color = Color.parseColor(colors[randomNumber]);
        return color;
    }
}

File: app/src/main/java/kiwi/slang/ColorsBackground.kt
package kiwi.slang

import android.content.Context
import java.util.Random

class ColorsBackground(private val context: Context) {

    private val colors: IntArray by lazy {
        val typedArray = context.resources.obtainTypedArray(R.array.background_colors)
        val colorsArray = IntArray(typedArray.length()) {
            typedArray.getColor(it, 0)
        }
        typedArray.recycle()
        colorsArray
    }

    internal val color: Int
        get() {
            val slangGenerator = Random()
            val randomNumber = slangGenerator.nextInt(colors.size)
            return colors[randomNumber]
        }
}

